instrumentation-api-semconv/src/test/java/io/opentelemetry/instrumentation/api/instrumenter/net/NetServerAttributesExtractorTest.java

```diff
@@ -129,7 +141,10 @@ void normal() {
             entry(SemanticAttributes.NET_SOCK_HOST_ADDR, "4:3:2:1::"),
             entry(SemanticAttributes.NET_SOCK_HOST_PORT, 8080L));
 
-    assertThat(endAttributes.build()).isEmpty();
+    assertThat(endAttributes.build())
+        .containsOnly(
+            entry(NetAttributes.NET_PROTOCOL_NAME, "http"),
+            entry(NetAttributes.NET_PROTOCOL_VERSION, "1.1"));
```

Comment:
is moving these attributes to onEnd needed? (could possibly impact sampling and attributes available for `http.server.active_requests`)

Reply:
It does make things simpler not to have separate getter interfaces for the client/server side (imagine `ServerClientGetter`/`ServerServerGetter` ...); and they're marked as sampling relevant in the semconv.

I figured that we can add some internal optimizations to the HTTP server extractors (by calling the getter with `response = null`) if this turns out to be an issue.

Reply:
> > they're marked as sampling relevant in the semconv
> 
> doesn't moving them to onEnd mean they're no longer available for sampling?

Uh, I lost a "not" 🙈 
Meant "they're not marked as sampling relevant in the semconv"

Reply:
https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/8695